R/env.R

```diff
@@ -0,0 +1,702 @@
+#' Get an environment.
+#'
+#' Environments are objects that create a scope for evaluation of R
+#' code. Reification of scope is one of the most powerful feature of
+#' the R language: it allows you to change what objects a function or
+#' expression sees when it is evaluated. In R, scope is hierarchical:
+#' each environment is defined with a parent environment. An
+#' environment and its grand-parents form together a linear
+#' hierarchy. All objects within the grand-parents are in scope unless
+#' they are eclipsed by synonyms (other bindings with the same names)
+#' in child environments.
+#'
+#' \code{env()} is a s3 generic. Methods are provided for functions,
+#' formulas and frames. If called with a missing argument, the
+#' environment of the current evaluation frame (see
+#' \code{\link{eval_stack}()}) is returned. If you call \code{env()}
+#' with an environment, it acts as the identity function and the
+#' environment is simply returned (this helps simplifying code when
+#' writing generic functions).
+#'
+#' \code{env_new()} creates a new environment. \code{env_next()}
```

Comment:
Also I find parent slightly more confusing because it suggests there can be several parents, whereas linked lists and environments are a purely linear sequence. Otoh the `inherit` terminology is a useful analogy and only makes sense with parents.